playbooks/common/openshift-cluster/upgrades/upgrade_control_plane.yml

```diff
@@ -231,6 +232,39 @@
     - reconcile_jenkins_role_binding_result.rc == 0
     when: (not openshift.common.version_gte_3_7 | bool) and (openshift.common.version_gte_3_4_or_1_4 | bool)
 
+  - when: (openshift.common.version_gte_3_6 | bool) and (not openshift.common.version_gte_3_7 | bool)
+    block:
+    - name: Retrieve shared-resource-viewer
+      oc_obj:
+        state: list
+        kind: role
+        name: "shared-resource-viewer"
+        namespace: "openshift"
+      register: objout
+
+    - name: Determine if shared-resource-viewer is protected
+      set_fact:
+        __shared_resource_viewer_protected: true
+      when: "'results' in objout and 'results' in objout['results'] and 'openshift.io/reconcile-protect' in objout['results']['results'][0]['metadata']['annotations'] and objout['results']['results']['metadata']['annotations']['openshift.io/reconcile-protect'] == 'true' }}"
+
+    - copy:
+        src: "{{ item }}"
+        dest: "/tmp/{{ item }}"
+      with_items:
+      - "{{ __master_shared_resource_viewer_file }}"
+      when: __shared_resource_viewer_protected is not defined
+
+    - name: Fixup shared-resource-viewer role
+      oc_obj:
+        state: present
+        kind: role
+        name: "shared-resource-viewer"
+        namespace: "openshift"
+        files:
+        - "/tmp/{{ __master_shared_resource_viewer_file }}"
+        delete_after: true
+      when: __shared_resource_viewer_protected is not defined
```

Comment:
We should emit some kind of warning when `__shared_resource_viewer_protected is defined` so the user has a way of knowing that they will need to reconcile this themselves somehow.

Reply:
I have no idea if there is a way to do that, @sdodson may know ?